cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -755,13 +751,13 @@ instance FromJSON ExecutionUnitPrices where
         <*> o .: "priceMemory"
 
 
-toAlonzoPrices :: ExecutionUnitPrices -> Maybe Alonzo.Prices
+toAlonzoPrices :: ExecutionUnitPrices -> Either String Alonzo.Prices
```

Comment:
I totally support this idea, however I suggest we do it as a separate follow up PR for two reasons:
* This PR is already pretty complex
* Right now those errors are local to this module only, but with changes that will expose those failures downstream,  as described in this comment: https://github.com/input-output-hk/cardano-node/pull/5013#discussion_r1154582800,  it would make even more sense to add types for each individual failure,

Reply:
Create an issue for this https://github.com/input-output-hk/cardano-node/issues/5092